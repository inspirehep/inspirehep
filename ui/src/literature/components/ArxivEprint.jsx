import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';

import ArxivEprintLink from './ArxivEprintLink';

class ArxivEprint extends Component {
  render() {
    const { eprint } = this.props;
    return (
      <span>
        <ArxivEprintLink>{eprint.get('value')}</ArxivEprintLink>
        {eprint.has('categories') && (
          <span> [{eprint.getIn(['categories', 0])}]</span>
        )}
      </span>
    );
  }
}

ArxivEprint.propTypes = {
  eprint: PropTypes.instanceOf(Map).isRequired,
};

export default ArxivEprint;
