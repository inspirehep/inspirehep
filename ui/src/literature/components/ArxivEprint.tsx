import React, { Component } from 'react';
import { Map } from 'immutable';

import ArxivEprintLink from './ArxivEprintLink';

type Props = {
    eprint: $TSFixMe; // TODO: PropTypes.instanceOf(Map)
};

class ArxivEprint extends Component<Props> {

  render() {
    const { eprint } = this.props;
    return (
      <span>
        <ArxivEprintLink>{eprint.get('value')}</ArxivEprintLink>
        {eprint.has('categories') && (
          <span> [{eprint.getIn(['categories', 0])}]</span>
        )}
      </span>
    );
  }
}

export default ArxivEprint;
