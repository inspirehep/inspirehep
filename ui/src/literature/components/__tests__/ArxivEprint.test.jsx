import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ArxivEprint from '../ArxivEprint';

describe('ArxivEprint', () => {
  it('renders with arXiv id', () => {
    const eprint = fromJS({
      value: '123.12345',
      categories: ['cat'],
    });
    const wrapper = shallow((
      <ArxivEprint
        eprint={eprint}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
