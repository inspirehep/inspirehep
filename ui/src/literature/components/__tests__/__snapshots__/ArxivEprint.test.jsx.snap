// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArxivEprint renders with arXiv id 1`] = `
<span>
  <ArxivEprintLink>
    123.12345
  </ArxivEprintLink>
  <span>
     [
    cat
    ]
  </span>
</span>
`;
